site/migration/01-migration.md

```diff
@@ -0,0 +1,257 @@
+## Migration from kpt `v0.39.0` to kpt `v1.0`
+
+- [What's new and changed?](#What's-new-and-changed?)
+  - [CLI changes](#CLI-changes)
+  - [Kptfile schema changes](#Kptfile-schema-changes)
+  - [`kpt pkg sync` deprecated](#kpt-pkg-sync-deprecated)
+  - [Changes to Functions](#Changes-to-Functions)
+    - [`kpt fn run` deprecated](#kpt-fn-run-deprecated)
+    - [Function Config](#Function-Config)
+    - [Function Results](#Function-Results)
+  - [Changes to Setters](#Changes-to-Setters)
+    - [Setter validations deprecated](#Setter-validations-deprecated)
+    - [Auto-setters deprecated](#Auto-setters-deprecated)
+    - [Function Results](#Function-Results)
+  - [Changes to Live](#changes-to-live)
+- [Migration steps](#Migration-steps)
+  - [Automated portion of migration](#Automated-portion-of-migration)
+    - [Changes made by the function](#Changes-made-by-the-function)
+  - [Manual portion of migration](#Manual-portion-of-migration)
+- [Timeline](#Timeline)
+
+kpt `v1.0` is going to be the latest major release of the kpt CLI. The implementation of kpt `v1.0`
+has changed considerably from kpt `v0.39.0`. A rich set of new features have been added
+as a result of the customers’ input and requirements. Some features have been
+deprecated or refactored in ways that make them incompatible with kpt `v0.39.0`.
+Since these are backwards incompatible changes, there should be a way for customers
+to migrate/fix their existing kpt packages which are compatible with `v0.39.0` version of kpt,
+to become compatible with kpt `v1.0`. This document outlines the end to end migration
+journey of customers using a comprehensive kpt package example.
```

Comment:
Somewhere at the top, link to the kpt book. Specifically, the user needs to read Chapter 1 and 2, i.e. the Quickstart and the concepts, specifically the workflows (https://kpt.dev/book/02-concepts/02-workflows) to be able to make sense of things. Trying to follow the migration doc (dealing with complexity of two versions) without understanding the basic middle of new version will lead to confusion.

Reply:
Updated Before you begin section.